SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs

```diff
@@ -419,97 +417,90 @@ public void CreateGroupParticipants(string token, List<MpNewParticipantDto> mpPa
 
-                if (nextAcEvent != null)
-                {
-                    // case # 1 - the current AC event's parent event is the current service event - check to 
-                    // see if there is another service event that day
-                    var nextServiceEvent = GetNextServiceEvent(nextAcEvent.CongregationId, participantEventMapDto.CurrentEvent.EventId);
+                    if (dailyEvents.Any(r => r.EventId != participantEventMapDto.CurrentEvent.EventId && r.ParentEventId == null))
+                    {
+                        nextServiceEvent = dailyEvents.First(r => r.EventId != participantEventMapDto.CurrentEvent.EventId && r.ParentEventId == null);
+                    }
 
-                    // sign them into a regular service
-                    if (nextServiceEvent == null || nextServiceEvent.EventId == participantEventMapDto.CurrentEvent.EventId)
+                    // Case #2 - no following service events, sign them into the current service event
+                    if (nextServiceEvent == null)
                     {
-                        returnEventIds.Add(participantEventMapDto.CurrentEvent.EventId);
-                        return returnEventIds;
+                        returnEvents.Add(dailyEvents.First(r => r.EventId == participantEventMapDto.CurrentEvent.EventId));
+                        return returnEvents;
                     }
 
-                    // if the event being signed into is an ac event and there is a later ac event,
-                    // sign them into the current service and the later ac event
-                    if (nextServiceEvent.EventId == participantEventMapDto.CurrentEvent.EventId)
+                    // Case #3 - no AC event for current event, but later AC, sign them into the current
+                    // event and later AC
+                    if (mpAcEventDto.ParentEventId != participantEventMapDto.CurrentEvent.EventId)
                     {
-                        returnEventIds.Add(participantEventMapDto.CurrentEvent.EventId);
-                        returnEventIds.Add(nextAcEvent.EventId);
-                        return returnEventIds;
+                        returnEvents.Add(dailyEvents.First(r => r.EventId == participantEventMapDto.CurrentEvent.EventId));
+                        returnEvents.Add(mpAcEventDto); 
+                        return returnEvents;
                     }
 
-                    // ReSharper disable once ConditionIsAlwaysTrueOrFalse -- JPC
-                    if (nextServiceEvent != null && nextAcEvent.ParentEventId == participantEventMapDto.CurrentEvent.EventId)
+                    // Case #4 - AC for current event and later event exists with no AC, sign them 
+                    // into the current AC event and later service event
+                    if (mpAcEventDto.ParentEventId == participantEventMapDto.CurrentEvent.EventId &&
+                        nextServiceEvent != null)
```

Comment:
If these cases are arranged correct then by the time we get here nextServiceEvent will never equal null so no need to check this.